web/content/assets/js/try.js

```diff
@@ -23,52 +37,107 @@ $(document).ready(function () {
         $.get("/jison/examples/"+file, function (data) {
                 $("#grammar").val(data);
                 $(document.body).removeClass("loading");
+                processGrammar();
             });
     });
-
+    
+    // recompile the grammar using a web worker,
+    // as the user types
+    var onChange = debounce(700, processGrammar);
+    $('#grammar').bind('input propertychange', onChange);
+    processGrammar();
 });
 
 function processGrammar () {
-    var type = "lalr";
-
-    var grammar = $("#grammar").val();
-    try {
-        var cfg = JSON.parse(grammar);
-    } catch(e) {
+    
+    function onError(e) {
+        console.log(e);
+        $("#gen_out").html("Oops. Make sure your grammar is " +
+            "in the correct format."+ "\n" + e.stack)
+            .removeClass('good')
+            .removeClass('warning')
+            .addClass('bad');
+    }
```

Comment:
Not strictly related to the web worker stuff, but I thought it would be nice if it printed out a full stack trace. More info is helpful. :)